spec/tests/check.fmf

```diff
@@ -16,6 +16,13 @@ description: |
 
     See :ref:`/plugins/test-checks` for the list of available checks.
 
+    The `result` key can be used to specify how the check result should be
+    interpreted. It accepts three options:
+    - 'respect' (default): The check result affects the overall test result.
+    - 'skip': The check result does not affect the overall test result.
```

Comment:
So here we're introducing a new value `skip` which is currently not present in the `Test.result` key specification. What about keeping the consistency here as well and support `pass, info, warn, error, fail` in the very same way?

Reply:
Hmmm,`skip` is definitely one of the values of https://github.com/teemtee/tmt/blob/main/tmt/result.py#L43, and it's also allowed by schema, https://github.com/teemtee/tmt/blob/main/tmt/schemas/common.yaml#L481. If it's not supposed to be supported as a `result` value, we should remove it from these places as well.

Reply:
Although, I'd argue the specification should be updated instead, because `skip` is a perfectly valid test outcome, like `pass` or `fail`, and it makes sense that one might want to enforce it as a test outcome via `result`, just like one might want to enforce `pass` or `xfail`.

Reply:
And +1 to accepting all keys allowed by `ResultInterpret`, i.e. why allowing `skip` and not `xfail` or `pass`. Except `custom`, that one does not make much sense for a test check.

Reply:
And `restraint`, `restraint` is also useless at this point of code :)

Reply:
Hm, looks more and more like we need `ChecResultInterpretat` enum & schema, too many exceptions.

Reply:
To be honest, I got pretty frustrated with this code. I guess I'm too dumb for mixing custom mixins, uninheritable classes, specifications, schemas, et al. I'm sure there are good reasons for all of this, but idk, somehow I think doing a relatively small change should be easier.

Anyway, should I create some BaseResultInterpretMixin class w/ SerializableContainer?

I tried to went back to using 'ignore', but now we're mixing two things together, results and how to interpret it. :(

Reply:
> To be honest, I got pretty frustrated with this code. I guess I'm too dumb for mixing custom mixins, uninheritable classes, specifications, schemas, et al. I'm sure there are good reasons for all of this, but idk, somehow I think doing a relatively small change should be easier.

That's sad to hear. I believe that while it might be a relatively small change, it's not a small change given the interaction of test and test check results. Schema, specification, the fact that enum inheritance is limited, well, that's a fact we can't bypass.

> Anyway, should I create some BaseResultInterpretMixin class w/ SerializableContainer?

I for one don't think so. Where do you think it would be needed?

> I tried to went back to using 'ignore', but now we're mixing two things together, results and how to interpret it. :(

We can get on a call and try to get through the parts that cause you trouble, if you want to.

Reply:
> I for one don't think so. Where do you think it would be needed?

It's just that the ResultsInterpret is not exactly [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).  
Only thing that came to mind, except for using str literals instead of enums.

> We can get on a call and try to get through the parts that cause you trouble, if you want to.

Thanks, much appreciated.
It seems like we strayed a bit too far from the original goal and while we have check result keys, it's used the same way as test result and I'm not sure if it is the best way to let users define that the check result should be "ignored".

Reply:
> I don't want to start rewriting the patch for you, but you obviously do not enjoy the coding at this moment, if it would help, we can sit on it.

Thanks @happz, I'll get to it with a pair of fresh eyes later. Sorry for letting my depression leak to the comments here.

Reply:
Ok, as you wish. Just a couple of notes:

* something like this could work for propagating how check results should be interpreted - `check_result = CheckResultInterpret(check.result.value)` is definitely not the right way how to get this info.
  ```diff
  -        return _result.interpret_result(invocation.test.result)
  +        return _result.interpret_result(
  +            invocation.test.result,
  +            {
  +                check.how: check.result
  +                for check in invocation.test.check
  +            }
  +        )
 
       def interpret_result(
               self,
  -            interpret: ResultInterpret) -> 'Result':
  +            interpret_test: ResultInterpret,
  +            interpret_checks: dict[str, CheckResultInterpret]) -> 'Result':
  ```
* use a helper function for setting a note, the repetition of `if`/`else` consumes 4 lines for every single note, a helper function would take care of `', ...'` and condition, greatly improving the readability of code.
* this looks like a mistake, check result propagated into result - is that correct? We probably don't want check result to be promoted to main result, instead failed results would turn the main result into a failed one - a skipped check does not mean main result should turn to `SKIP`
   ```python
              elif interpret_check in [CheckResultInterpret.PASS, CheckResultInterpret.FAIL,
                                    CheckResultInterpret.INFO, CheckResultInterpret.WARN,
                                    CheckResultInterpret.ERROR, CheckResultInterpret.SKIP]:
                  self.result = ResultOutcome(check.value)
  ```
* the order should be probably reversed: first we need to sort out check results, because they might be `PASS`, but their `result` key might be set to `fail`, turning them to failed checks - `any(check.result == ResultOutcome.FAIL for check in self.check)` will miss checks that would turn into failed checks because of their interpretation.